Helper/FeatureSwitch/Manager.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Helper\FeatureSwitch;
+
+
+use Magento\Framework\App\Helper\Context;
+use Bolt\Boltpay\Helper\GraphQL\Client as GQL;
+use Bolt\Boltpay\Model\FeatureSwitchRepository;
+use Magento\Framework\App\Helper\AbstractHelper;
+
+class Manager extends AbstractHelper {
+
+    /**
+     * @var GQL
+     */
+    private $gql;
+
+    /**
+     * @var fsRepo
+     */
+    private $fsRepo;
+
+    /**
+     * @param Context $context
+     * @param GQL $gql
+     * @param FeatureSwitchRepository $fsRepo
+     * @codeCoverageIgnore
+     */
+    public function __construct(
+        Context $context,
+        GQL $gql,
+        FeatureSwitchRepository $fsRepo
+    ) {
+        parent::__construct($context);
+        $this->gql = $gql;
+        $this->fsRepo = $fsRepo;
+    }
+
+    /**
+     * This method gets feature switches from Bolt and updates the local DB with
+     * the latest values. To be used in upgrade data and webhooks.
+     *
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function updateSwitchesFromBolt() {
+        $switchesResponse = $this->gql->getFeatureSwitches();
+
+        $switches = array();
+        if ($switchesResponse->getData()) {
+            $data = $switchesResponse->getData();
+            if (isset($data["response"])
+                && isset($data["response"]->data)
+                && isset($data["response"]->data->plugin)
+                && isset($data["response"]->data->plugin->features)
+            ) {
+                $switches = $data["response"]->data->plugin->features;
```

Comment:
you could also do

```
@$switches = $data["response"]->data->plugin->features;
```

it will work like a optional chain (@ is error control operator https://www.php.net/manual/en/language.operators.errorcontrol.php )

Reply:
`@` only works on expressions, need it on the RHS if you want to do this approach. This is probably fine - you are checking on `switches` directly after this.
```
$switches = @$data["response"]->data->plugin->features;
```

Reply:
WOW. apparently this is the stfu operator! LOVE IT